From d2361b89fefb07b669b4cf67257fc3bf06afd9f5 Mon Sep 17 00:00:00 2001
From: Nicklas Larsson <n_larsson@yahoo.com>
Date: Mon, 18 Sep 2023 20:12:18 +0200
Subject: [PATCH] BUG: define SWIGPY_SLICEOBJECT

SWIG v 4.1.0+ use PyObject instead of PySliceObject after
https://github.com/swig/swig/commit/e683168018558f4e73dc26629c4bbec224d23438

This adds the definition of SWIGPY_SLICEOBJECT.
---
 Wrapping/Generators/Python/module_ext.i.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Wrapping/Generators/Python/module_ext.i.in b/Wrapping/Generators/Python/module_ext.i.in
index 53f05530135..826d336348a 100644
--- a/Wrapping/Generators/Python/module_ext.i.in
+++ b/Wrapping/Generators/Python/module_ext.i.in
@@ -6,8 +6,10 @@
 
 #if PY_VERSION_HEX >= 0x03020000
 # define SWIGPY_SLICE_ARG(obj) ((PyObject*) (obj))
+# define SWIGPY_SLICEOBJECT PyObject
 #else
 # define SWIGPY_SLICE_ARG(obj) ((PySliceObject*) (obj))
+# define SWIGPY_SLICEOBJECT PySliceObject
 #endif
 %}
 
